### PR TITLE
Update to LDC 1.15.0 stable release, incl. LLVM 8.0.0

### PR DESCRIPTION
**Pull request by kinke**
This is a dummy PR, just for making the 1.15 branch visible. Pinging @WebDrake.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -39,6 +39,7 @@ parts:
         -DLLVM_ROOT_DIR=../../llvm/install \
         -DBUILD_LTO_LIBS=ON \
         -DLDC_INSTALL_LTOPLUGIN=ON \
+        -DLDC_INSTALL_LLVM_RUNTIME_LIBS=ON \
```
Comment (kinke) [inline snap/snapcraft.yaml:42]: This makes sure the LLVM compiler-rt libraries are included; they are needed for PGO, sanitizers etc. These libs are affected by https://github.com/ldc-developers/ldc/issues/3054 though as LLVM currently isn't compiled with `-Wa,-mrelax-relocations=no`.

**Comment by WebDrake** [comment]
Thanks for the patch.  A couple of notes for future reference:

* the branch name actually matches what I use for the individual release branches which packages are built from

* I know it's just for testing, but we really shouldn't be submitting version upgrade PRs against the 1.14 branch!

What I would usually do is:

* create a `1.N+1` branch from the `1.N` branch

* submit the version upgrade patch against the `1.N+1` branch

Anyway, per se this LGTM apart from one small thing -- wasn't there an extra build optimization that could be enabled with 1.15 ... ?  I remember you mentioning it when recommending the bootstrap compiler be built from the same version as the release itself.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (WebDrake) [reply snap/snapcraft.yaml:42]: Is that something we can address on the package side?  Bearing in mind that we build LLVM in the package?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -108,6 +107,8 @@ parts:
     - -DLLVM_BINUTILS_INCDIR=/usr/include
     - -DLLVM_TARGETS_TO_BUILD=AArch64;ARM;Mips;MSP430;NVPTX;PowerPC;X86
     - -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=RISCV;WebAssembly
+    - -DLLVM_ENABLE_TERMINFO=OFF
+    - -DLLVM_ENABLE_LIBEDIT=OFF
```
Comment (WebDrake) [inline snap/snapcraft.yaml:111]: The LLVM 8 fork adds `SPIRV` support.  Is that something we can/should add an extra target for?

Comment (kinke) [reply snap/snapcraft.yaml:111]: It's a separate extra library, src included as git submodule, and built automatically, no conventional LLVM target.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (kinke) [reply snap/snapcraft.yaml:42]: Sure, we'd simply need that extra C[++] flag when building LLVM too. I just don't want to override the default flags and am unsure as to whether setting `CMAKE_C[XX]_FLAGS` would override them.

**Comment by kinke** [comment]
> I know it's just for testing

Wouldn't even by necessary, all branches are tested too. It was just to make this visible to you. ;)

> wasn't there an extra build optimization

Nope, but v1.15 as host compiler (with the GC improvements in its druntime) significantly increases the usefulness of the new `-lowmem` option, so we're already set.

**Comment by WebDrake** [comment]
> Nope, but v1.15 as host compiler (with the GC improvements in its druntime) significantly increases the usefulness of the new `-lowmem` option, so we're already set.

So no extra build flags needed, the benefits are automatic?

**Comment by kinke** [comment]
Yes, just a better/less wasteful GC in druntime.

**Comment by WebDrake** [comment]
Awesome.  In that case I'll close this PR and just set up the job to build this version.  We can deal with the LLVM C++ flags issue in a separate PR.

**Comment by WebDrake** [comment]
Closing, as the branch should not be merged into 1.14.

**Comment by WebDrake** [comment]
Snap package definition in place: https://code.launchpad.net/~ldc-developers/+snap/ldc2-1.15

**Pull request closed without merging**